modules/agar/src/main/ts/ephox/agar/mouse/Clicks.ts

```diff
@@ -1,8 +1,86 @@
-import { Document, HTMLElement, MouseEvent, window } from '@ephox/dom-globals';
-import { Element, Location } from '@ephox/sugar';
+import { Node as DomNode, Document, HTMLElement, MouseEvent, window } from '@ephox/dom-globals';
+import { Element, Location, Node, Position } from '@ephox/sugar';
+import { Fun, Obj } from '@ephox/katamari';
 
-const LEFT_CLICK = 0;
-const RIGHT_CLICK = 2;
+// The 'button' field of the mouse event - which button was pressed to create the event. Pick only one value. Not defined for mouseenter,
+// mouseleave, mouseover, mouseout or mousemove.
+const button = {
```

Comment:
Not really an issue given this is a testing library, but doing this is an antipattern and should be avoid as you can't treeshake unused properties. It's better to do something like this instead:
```
const LEFT_CLICK = 0;
const MIDDLE_CLICK = 1;
const RIGHT_CLICK = 2;

const LEFT_CLICK_BIT = 1;
const RIGHT_CLICK_BIT = 2;
const MIDDLE_CLICK_BIT = 4;
```

It'd be good if we could clean that up to avoid potential confusion in the future.

Reply:
You could also just use prefixes if you prefer, eg `BUTTON_*` and `BUTTONS_*`.

Reply:
Actually given you export this later, for naming consistency, it might be better to use camel case, eg: `leftClickButton` and `leftClickButtons`.